### ISSUE TITLE
MGM mode: Possibly take first byte of IV modulo 128 during init

### ISSUE THREAD

**Issue by alex7691076** [gost-engine/engine#261]
Is there any reason why you guys didn't just take first byte of given IV during MGM init modulo 128?

https://github.com/gost-engine/engine/blob/361dcb32d8a4b439cdb3cbb85ebd2eebc30f9fbf/gost_grasshopper_cipher.c#L265

This makes it much easier to work with by just giving 16 random bytes, instead of having to supply special 127 bit IV just for MGM to avoid getting back the GOST_R_INVALID_IV_LENGTH error.

**Comment by beldmit** [comment]
No, there is no special reason here. It was quick-and-dirty code to be fixed the way you suggest.
